pkg/cluster/k8sres.go

```diff
@@ -219,8 +219,27 @@ PatroniInitDBParams:
 	if patroni.TTL != 0 {
 		config.Bootstrap.DCS.TTL = patroni.TTL
 	}
-	if patroni.Slots != nil {
-		config.Bootstrap.DCS.Slots = patroni.Slots
+
+	// The SPILO image is taking in account the 'Slots' settings **only** when bootstraping a new cluster.
+	// Therefore, updating this value in the SPILO_CONFIGURATION of the statefulset is pointless, but still
+	// required at cluster creation.
+	if isNewCluster {
+		if patroni.Slots != nil {
+			config.Bootstrap.DCS.Slots = patroni.Slots
+		}
+	} else {
+		oldestKnownPatroniCfg := acidv1.Patroni{}
+		if initialPatroniConfig == "" {
+			// when upgrading postgres-operator to this new version, existing StatefulSet won't have this annotation, so we're using the current specs
+			oldestKnownPatroniCfg = *patroni
+		} else {
+			if err := json.Unmarshal([]byte(initialPatroniConfig), &oldestKnownPatroniCfg); err != nil {
+				logger.Errorf("cannot decode initial patroni config from annotations: %s (%v)", err, initialPatroniConfig)
+				return "", err
+			}
+		}
+
+		config.Bootstrap.DCS.Slots = oldestKnownPatroniCfg.Slots
```

Comment:
but at this point the cluster already exists and bootstrap options have no effect. why are we resetting it again ? to get the same stateful set spec ?